tracer/src/Datadog.Trace/OpenTelemetry/Sdk.cs

```diff
@@ -0,0 +1,57 @@
+// <copyright file="Sdk.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Activity.DuckTypes;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.OpenTelemetry
+{
+    internal static class Sdk
+    {
+        public static void Initialize()
+        {
+            // Initialize no-op OpenTelemetry API implementations that the OpenTelemetry SDK would typically overwrite
+            // Note: Since this initialization is run near the time of application startup, any further updates by the
+            //       OpenTelemetry SDK or user should overwrite these settings
+            var propagatorsType = Type.GetType("OpenTelemetry.Context.Propagation.Propagators, OpenTelemetry.Api", throwOnError: false);
+            if (propagatorsType is not null
+                && propagatorsType.Assembly.GetName().Version >= new Version(1, 0, 0))
+            {
+                var defaultTextMapPropagatorProperty = propagatorsType.GetProperty("DefaultTextMapPropagator", BindingFlags.Static | BindingFlags.NonPublic | BindingFlags.Public);
+                var setMethod = defaultTextMapPropagatorProperty?.GetSetMethod(true);
```

Comment:
Any reason why we use `defaultTextMapPropagatorProperty.GetSetMethod()` here and then `setMethod.Invoke()` below, instead of skipping the `setMethod` and calling `defaultTextMapPropagatorProperty.SetValue()` below?

I've this pattern in the parametric tests as well and I don't know if it was intentional 😅 
https://github.com/DataDog/system-tests/blob/2cb5033072c1813eccdce332153a378607e73d85/utils/build/docker/dotnet/parametric/Endpoints/ApmTestApi.cs#L51

Reply:
I didn't try `SetValue` (maybe I should test locally to be sure), but the [API docs](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.propertyinfo.setvalue?view=net-9.0#system-reflection-propertyinfo-setvalue(system-object-system-object)) indicate that `SetValue` may throw a `MethodAccessException` for private/protected methods so I wanted to avoid that. Let me get back to you on this question

Reply:
`MethodBase.Invoke` can also throw a `MethodAccessException` ([API docs](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.methodbase.invoke)), so I think it's the same risk either way 😅

Reply:
I tested `defaultTextMapPropagatorProperty.SetValue()` locally and it works, so I've made the change in a9260144f943a2b0babfa4363f0cdb15e4554b00 . Thanks for the feedback!

Reply:
Thanks!